src/azul/terra.py

```diff
@@ -201,10 +206,25 @@ class TDRClient(SAMClient):
     A client for the Broad Institute's Terra Data Repository aka "Jade".
     """
 
+    def project_for_source(self, source: TDRSource) -> str:
+        """
+        Use the TDR API to find which Google Cloud project contains the source.
+        """
+        response = self._get_source_response(source)
+        assert response.status == 200
```

Comment:
Generally, asserts are for caching bugs or enforcing invariants. If an assert fires, one of us made a mistake. Another service returning an error does not fall into that category. You can use `require` instead.